src/OpenTelemetry.Instrumentation.AspNetCore/MeterProviderBuilderExtensions.cs

```diff
@@ -31,20 +34,56 @@ public static class MeterProviderBuilderExtensions
         /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
         public static MeterProviderBuilder AddAspNetCoreInstrumentation(
             this MeterProviderBuilder builder)
+            => AddAspNetCoreInstrumentation(builder, name: null, configureAspNetCoreInstrumentationOptions: null);
+
+        /// <summary>
+        /// Enables the incoming requests automatic data collection for ASP.NET Core.
+        /// </summary>
+        /// <param name="builder"><see cref="MeterProviderBuilder"/> being configured.</param>
+        /// <param name="configureAspNetCoreInstrumentationOptions">Callback action for configuring <see cref="AspNetCoreMetricsInstrumentationOptions"/>.</param>
+        /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
+        public static MeterProviderBuilder AddAspNetCoreInstrumentation(
+            this MeterProviderBuilder builder,
+            Action<AspNetCoreMetricsInstrumentationOptions> configureAspNetCoreInstrumentationOptions)
+            => AddAspNetCoreInstrumentation(builder, name: null, configureAspNetCoreInstrumentationOptions);
+
+        /// <summary>
+        /// Enables the incoming requests automatic data collection for ASP.NET Core.
+        /// </summary>
+        /// <param name="builder"><see cref="MeterProviderBuilder"/> being configured.</param>
+        /// <param name="name">Name which is used when retrieving options.</param>
+        /// <param name="configureAspNetCoreInstrumentationOptions">Callback action for configuring <see cref="AspNetCoreMetricsInstrumentationOptions"/>.</param>
+        /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
+        public static MeterProviderBuilder AddAspNetCoreInstrumentation(
+            this MeterProviderBuilder builder,
+            string name,
+            Action<AspNetCoreMetricsInstrumentationOptions> configureAspNetCoreInstrumentationOptions)
         {
             Guard.ThrowIfNull(builder);
 
-            // TODO: Implement an IDeferredMeterProviderBuilder
+            name ??= Options.DefaultName;
+
+            if (configureAspNetCoreInstrumentationOptions != null)
+            {
+                builder.ConfigureServices(services => services.Configure(name, configureAspNetCoreInstrumentationOptions));
+            }
+
+            builder.ConfigureBuilder((sp, builder) =>
+            {
+                var options = sp.GetRequiredService<IOptionsMonitor<AspNetCoreMetricsInstrumentationOptions>>().Get(name);
+
+                // TODO: Implement an IDeferredMeterProviderBuilder
```

Comment:
nit: Do me a favor and remove this TODO the ConfigureBuilder call satisfies IDeferredMeterProviderBuilder.

Reply:
Removed